keylime/tpm/tpm_main.py

```diff
@@ -1200,6 +1270,44 @@ def check_quote(
             hash_alg,
         )
 
+    def check_quote_timing(self, previous_clockinfo, quote):
+        # Sanity check quote clock information
+
+        current_clockinfo = {}
+
+        if previous_clockinfo:
+            retout, success = self._tpm2_printquote(quote, False)
+            if not success:
+                return False, "tpm2_print failed with " + str(retout), current_clockinfo
+
+            current_clockinfo = config.yaml_to_dict(retout, logger=logger)["clockInfo"]
+
+            if "clock" in previous_clockinfo and "clock" in current_clockinfo:
+                resetdiff = current_clockinfo["resetCount"] - previous_clockinfo["resetCount"]
+                restartdiff = current_clockinfo["restartCount"] - previous_clockinfo["restartCount"]
+
+                if resetdiff < 0:
+                    return "resetCount value decreased on TPM between two consecutive quotes", False, current_clockinfo
+
+                if restartdiff < 0:
```

Comment:
This is very interesting. Can this happen?

Reply:
I imagine that this would happen only in case of a "malformed" or "corrupted" quote.

Reply:
Maybe add a comment what this checks and also for the other sanity checks.

Reply:
Fair, will do.